### PR TITLE
[WIP] Add cgal

### PR DESCRIPTION
**Pull request by massich**

### CODE REVIEW

**Comment by massich** [comment]
```sh
$ ctest -v . -R cgal
Test project /home/sik/code/openmeeg/build
    Start 262: cgal_mesh_function
1/3 Test #262: cgal_mesh_function ...............   Passed    0.05 sec
    Start 263: cgal_remesh
2/3 Test #263: cgal_remesh ......................   Passed    1.47 sec
    Start 264: cgal_decimate
3/3 Test #264: cgal_decimate ....................   Passed    0.14 sec

100% tests passed, 0 tests failed out of 3

Total Test time (real) =   1.70 sec

```

**Comment by massich** [comment]
@agramfort do you know why brew complains that `cgal` is already [installed](https://travis-ci.org/openmeeg/openmeeg/jobs/357354069#L2381)?

If its installed only once in `.travis.yml`? 
```sh
❯ cat ../.travis.yml| grep -i cgal | grep brew
        brew install cgal;
```

File: .travis.yml [hunk 1]
```diff
@@ -97,6 +83,21 @@ before_install:
...
+  ############################################################################
+  # Install CGAL
+  ############################################################################
+  - if [[ "${USE_CGAL}" == "ON" ]]; then
+        if [[ "${TRAVIS_OS_NAME}" == "linux" ]]; then
+            if ! [ -x "$(command -v conda)" ]; then
+                git clone https://github.com/astropy/ci-helpers.git;
+                source ci-helpers/travis/setup_conda.sh;
+            fi
+            conda install -c conda-forge libmatio;
```
Comment (agramfort) [inline .travis.yml]: did you mean conda install cgal?

Comment (massich) [reply .travis.yml]: sure

**Comment by massich** [comment]
I started installing it using conda, but we can use [this ppa](https://github.com/nextgis/ppa/tree/master/cgal/debian) so that we don't have to a) `dist upgrade` which is a bad idea, or b) [pinning](https://help.ubuntu.com/community/PinningHowto) [`libcgal12`](https://launchpad.net/ubuntu/+source/cgal)

**Comment by agramfort** [comment]
give it a try

**Comment by massich** [comment]
5b88bb57 * @ improve travis testing, adding cases with vtk/cgal
00844e0b * packaging: switch back to vecLib standalone on mac and clapack on linux.

**Comment by massich** [comment]
Apparently, cgal was added in #154 taken out in 00844e0b, put back in #258 and out again when we migrated the building system in #268

**Pull request closed without merging**